File: version_java/src/main/java/com/example/version_java/data/repo/CriminalRepository.java
package com.example.version_java.data.repo;

import com.example.version_java.room.entity.CriminalEntity;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import kotlin.jvm.functions.Function1;

public interface CriminalRepository {

    void getRemoteCriminals(
            @NotNull Function1 onSuccess,
            @NotNull Function1 onFailure
    );

    void getLocalCriminals(
            @NotNull Function1 onSuccess,
            @NotNull Function1 onFailure
    );

    void registerCriminalEntity(
            ArrayList<CriminalEntity> entityList,
            @NotNull Function1 result
    );

    void getCriminalEntity(
            @NotNull String name,
            @NotNull Function1 onSuccess,
            @NotNull Function1 onFailure
    );
}

File: app/src/main/java/com/example/crimenotification/data/repo/CriminalRepository.kt
package com.example.crimenotification.data.repo

import com.example.crimenotification.data.model.CriminalItem
import com.example.crimenotification.network.response.CriminalResponse
import com.example.crimenotification.room.entity.CriminalEntity
import com.example.crimenotification.util.Result

interface CriminalRepository {

    suspend fun getRemoteCriminals(): Result<List<CriminalResponse>>

    suspend fun getLocalCriminals(): Result<List<CriminalEntity>>

    suspend fun registerCriminalEntity(entityList: List<CriminalEntity>): Boolean
}